### ISSUE TITLE
Add style guide

### ISSUE THREAD

**Issue by johnerikhalse** [iipc/openwayback#295]
There should be a guide for coding style on the wiki. This style should preferably be enforced in the build, but since that requires a lot of changes in the code, the first step should be to report style violations.

**Comment by johnerikhalse** [comment]
I have added a style guide. Consider it as a work in progress. Please discuss and enhance it.
https://github.com/iipc/openwayback/wiki/Style-guide

**Comment by kris-sigur** [comment]
I'm OK with the current draft. The only point I'd question is

> All classes and methods should have Javadoc comments. 

That is a bit on the optimistic side. Perhaps a slightly less ambitious clause would be warranted

> All classes must have a Javadoc comment. All public and protected methods should have a Javadoc comment. The use of Javadoc comments for other methods is encouraged, especially if they are more than 5 lines long.

**Comment by ldko** [comment]
I think these are sensible conventions in the current draft. I also think the more specific clause about Javadoc comments would be helpful to include.

**Comment by johnerikhalse** [comment]
I agree. Document is updated. I also added a line about avoiding the `@author` tag. I think `git blame` is a better alternative.
